.github/workflows/prepare-release-branch.yml

```diff
@@ -28,14 +28,15 @@ jobs:
       - uses: actions/checkout@v3
 
       - name: Create release branch
+        shell: bash {0}
         run: |
           version=$(.github/scripts/get-version.sh)
-          if [[ $version =~ ^([0-9]+)\.([0-9]+)\.0$ ]]; then
+          if [[ $version =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
             release_branch_name="release/v${version}"
-            git ls-remote --exit-code --heads origin refs/heads/$release_branch_name
-            if [ "$?" == "0" ] ; then 
+            release_branch_exists=$(git ls-remote --exit-code --heads origin refs/heads/$release_branch_name)
+            if [[ $release_branch_exists != "" ]] ; then
```

Comment:
what was the issue here? (so we can fix in other repos also if needed)

Reply:
I don't think we have that `git ls-remote` check in other repos; probably the shell used `set -e` and `git ls-remote` returned non-zero exit code

Reply:
This is a check that only exists in this repo. The other repos have a process in which the version is automatically incremented which is reflected in the release branch. In this repo, its much easier to accidentally run the release workflow against main without having manually bumped the `semantic-conventions` release version. This check helps avoid that.

Reply:
> I don't think we have that git ls-remote check in other repos; probably the shell used set -e and git ls-remote returned non-zero exit code

Exactly, but I pushed a commit to avoid the non-zero exit code, as I mentioned [here](https://github.com/open-telemetry/semantic-conventions-java/pull/17#discussion_r1313171543).

Figure its safer to fail on non-zero exit codes if possible.